TPPDF/Classes/PDFGenerator.swift

```diff
@@ -260,9 +270,26 @@ open class PDFGenerator  {
             y = contentHeight + maxHeaderHeight() + headerSpace
         }
         
-        let frame = CGRect(x: x, y: y, width: aspectWidth, height: aspectHeight)
+        // resize
+        let resizeFactor = (3 * imageQuality > 1) ? 3 * imageQuality : 1
+        let resizeImageSize = CGSize(width: aspectWidth * resizeFactor, height: aspectHeight * resizeFactor)
+        UIGraphicsBeginImageContext(resizeImageSize)
+        image.draw(in: CGRect(x:0, y:0, width: resizeImageSize.width, height: resizeImageSize.height))
+        var compressedImage = UIGraphicsGetImageFromCurrentImageContext()
+        UIGraphicsEndImageContext()
+        
+        // compression
+        if let image = compressedImage, let jpegData = UIImageJPEGRepresentation(image, imageQuality) {
```

Comment:
Maybe we should use a guard statement and some kind of error handling in case something goes wrong.

One way would be creating an error enum `TPPDFError`